src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1735,21 +1736,12 @@ fn resolution_failure(
                         }
 
                         if !path_str.contains("::") {
-                            if disambiguator.map_or(true, |d| d.ns() == MacroNS)
-                                && let Some(&res) = collector.cx.tcx.resolutions(()).all_macro_rules
-                                                             .get(&Symbol::intern(path_str))
-                            {
-                                diag.note(format!(
-                                    "`macro_rules` named `{path_str}` exists in this crate, \
-                                     but it is not in scope at this link's location"
```

Comment:
Can you keep this error please? So this only removes the `recovered_res` but still gives a helpful diagnostic?